core/src/main/java/org/elasticsearch/gateway/MetaDataStateFormat.java

```diff
@@ -279,7 +280,7 @@ public  T loadLatestState(ESLogger logger, Path... dataLocations) throws IOExcep
         //       new format (ie. legacy == false) then we know that the latest version state ought to use this new format.
         //       In case the state file with the latest version does not use the new format while older state files do,
         //       the list below will be empty and loading the state will fail
-        for (PathAndStateId pathAndStateId : Collections2.filter(files, new StateIdAndLegacyPredicate(maxStateId, maxStateIdIsLegacy))) {
+        for (PathAndStateId pathAndStateId : files.stream().filter(new StateIdAndLegacyPredicate(maxStateId, maxStateIdIsLegacy)).collect(Collectors.toCollection(ArrayList::new))) {
```

Comment:
can you put this statement on several lines (like you did elsewhere) to improve readability?